waku/v2/node/wakunode2.nim

```diff
@@ -468,26 +468,29 @@ proc mountStore*(node: WakuNode, store: MessageStore = nil, persistMessages: boo
   node.switch.mount(node.wakuStore, protocolMatcher(WakuStoreCodec))
     
 when defined(rln):
-  proc addRLNRelayValidator*(node: WakuNode, pubsubTopic: string) =
+  proc addRLNRelayValidator*(node: WakuNode, pubsubTopic: string, contentTopic: ContentTopic) =
     ## this procedure is a thin wrapper for the pubsub addValidator method
     ## it sets message validator on the given pubsubTopic, the validator will check that
     ## all the messages published in the pubsubTopic have a valid zero-knowledge proof 
     proc validator(topic: string, message: messages.Message): Future[pubsub.ValidationResult] {.async.} =
       let msg = WakuMessage.init(message.data) 
       if msg.isOk():
-        let 
-          wakumessage = msg.value()
-          # validate the message
-          validationRes = node.wakuRlnRelay.validateMessage(wakumessage)
+        let wakumessage = msg.value()
+        # check the contentTopic
+        if (wakumessage.contentTopic != "") and (wakumessage.contentTopic != contentTopic):
+          info "content topic did not match:", contentTopic=wakumessage.contentTopic, payload=string.fromBytes(wakumessage.payload)
+          return pubsub.ValidationResult.Accept
+        # validate the message
+        let validationRes = node.wakuRlnRelay.validateMessage(wakumessage)
         case validationRes:
           of Valid:
-            info "message validity is verified, relaying:", wakumessage=wakumessage
+            info "message validity is verified, relaying:", wakumessage=wakumessage, payload=string.fromBytes(wakumessage.payload)
```

Comment:
Does this mean that with RLN enabled we'll log at `info` level for each message? Per-message logs should preferably be at `TRC` level to help keep logs useful.

Reply:
Sure, I'll change the log level to trace or debug

Reply:
done in https://github.com/status-im/nim-waku/pull/835/commits/ea4c9a6a9df9a27aae99e1dfec0279106c754b0b